book/src/components/resulttable.md

```diff
@@ -0,0 +1,75 @@
+| ------------------ | --------- | ------- | --------------------------------------------------------------- |
+| `title`            | `string`  | `""`    | Optional title displayed above the table.                       |
+| `pageSize`         | `number`  | `10`    | Number of rows per page.                                        |
+| `pageSizeSwitcher` | `boolean` | `false` | If true, enables a dropdown to change the number of rows shown. |
+
+---
+
+## Slots
+
+| Slot Name                      | Description                                    |
+| ------------------------------ | ---------------------------------------------- |
+| `lens-result-above-pagination` | Renders content above the pagination controls. |
+| `beneath-pagination`           | Renders content below the pagination controls. |
+
+---
+
+## CSS Parts
+
+| Part Name                                    | Description                                                              |
+| -------------------------------------------- | ------------------------------------------------------------------------ |
+| `lens-result-table-title`                    | The title heading above the table.                                       |
+| `lens-result-table`                          | The table element itself.                                                |
+| `lens-result-table-header`                   | The `<thead>` element.                                                   |
+| `lens-result-table-header-row`               | The header row `<tr>`.                                                   |
+| `lens-result-table-header-cell`              | Generic header cells.                                                    |
+| `lens-result-table-header-cell-checkbox`     | Header cell containing the "select all" checkbox.                        |
+| `lens-result-table-header-datatype`          | Header cells for data type columns (sortable, may include info buttons). |
+| `lens-result-table-header-checkbox`          | The checkbox used to select all rows.                                    |
+| `lens-result-table-table-body`               | The `<tbody>` containing the data rows.                                  |
+| `lens-result-table-pagination`               | Container for pagination controls.                                       |
+| `lens-result-table-pagination-button`        | Previous and next arrow buttons.                                         |
+| `lens-result-pagination-pagination-previous` | Specifically styles the "previous" button.                               |
+| `lens-result-pagination-pagination-next`     | Specifically styles the "next" button.                                   |
+| `lens-result-table-pagination-pagenumber`    | Displays the current page number and total page count.                   |
+| `lens-result-table-pagination-switcher`      | Container for the page size switcher dropdown.                           |
+
+---
+
+## Example
```

Comment:
Example section should come earlier to be consistent with other component doc pages